### ISSUE TITLE
stdstar out of memory with 6 exposures

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#1785]
Tile 82266 on night 20210928 was observed 6 times (expids 102103-102108).  The stdstar joint fit across all these exposures runs out of memory:
```
srun -N 1 -n 30 -c 4 --cpu-bind=cores  desi_proc_joint_fit   --obstype science --cameras a0123456789 -n 20210928 -e 102103,102104,102105,102106,102107,102108 --mpi --mpistdstars
...
slurmstepd: error: Detected 1 oom-kill event(s) in StepId=59371624.0. Some of your processes may have been killed by the cgroup out-of-memory handler.
srun: error: nid10540: tasks 3,13,15,23,25: Out Of Memory
srun: launch/slurm: _step_signal: Terminating StepId=59371624.0
FAILED: done at Tue May 24 18:19:57 PDT 2022
```

Unmerged PR #1735 also identified this, but addressed this by increasing the number of nodes for all stdstar jobs even though that isn't needed for the majority of tiles.  Let's try to fix this by adjusting the actual amount of memory used, e.g. by trimming the input frames to just the standard stars instead of keeping the xentire frame in memory.  This would require some bookkeeping updates to make sure that the output stdstar indices are correct relative to the original input frames and not just the trimmed frames.

Also consider reviving old PR #1284 to do I/O on a single rank before broadcasting, which might be an even better place to trim the frames down.

**Comment by sbailey** [comment]
night 20220324 tile 82636 with 8 exposures is another case of this stdstar memory problem.

**Comment by sbailey** [comment]
Implemented in PR #1820; closing.
